### PR TITLE
Proposal of a CocoaPods support + custom JSON*able nested types

### PR DESCRIPTION
**Pull request by MonsieurDart**

### CODE REVIEW

**Comment by Liquidsoul** [comment]
Thanks, I'll look into it as soon as I can 😉

File: README.md [hunk 1]
```diff
@@ -114,12 +115,70 @@ struct Contact {
...
+Configure your `.sourcery.yml` file with this:
+
+``` yaml
+[…]
+templates:
+  - Pods/Sourcery-AutoJSONSerializable/Templates/
+  […]
+```
+
+Add those protocoles anywhere in your source code:
```
Comment (Liquidsoul) [inline README.md]: typo here: `protocols`

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+MIT License
+
+Copyright (c) 2017 Sébastien Duperron @liquidseb
```
Comment (Liquidsoul) [inline LICENSE]: `@liquidseb` is not relevant here. It should be removed or replaced by my GitHub handle `@liquidsoul`

File: README.md [hunk 3]
```diff
@@ -83,11 +83,12 @@ Making it conform to AutoJSONDeserializable would result make it support this JS
 
 # Features #
 
-  * primitive JSON types (String, Int, Double)
+  * Primitive JSON types (String, Int, Double).
   * ISO8601 formatted dates
```
Comment (Liquidsoul) [inline README.md]: The final dot should be put on every line here or on none. If we put dots, let's not forget this little one here 😉

File: README.md [hunk 4]
```diff
@@ -114,12 +115,70 @@ struct Contact {
...
+[…]
+templates:
+  - Pods/Sourcery-AutoJSONSerializable/Templates/
+  […]
+```
+
+Add those protocoles anywhere in your source code:
+
+``` swift
+protocol AutoJSONSerializable {}
```
Comment (Liquidsoul) [inline README.md]: Just a side note here, I'm thinking of removing the need of these protocols altogether and use a Sourcery annotations instead:
```swift
// sourcery: AutoJSONSerializable
struct Contact {
    [...]
}
```

File: README.md [hunk 5]
```diff
@@ -114,12 +115,70 @@ struct Contact {
...
+
+``` swift
+struct Contact, AutoJSONSerializable {
+    [...]
+}
+```
+
+## Manual Install ##
+
+The current installation process is pretty basic and does not support any dependency managers yet.
```
Comment (Liquidsoul) [inline README.md]: Let's remove this sentence, it won't have any sense one this is merged.

File: README.md [hunk 6]
```diff
@@ -114,12 +115,70 @@ struct Contact {
...
+
+protocol JSONDeserializable {
+    init?(JSONObject: [String: Any])
+}
+```
+
+And, finally, to use it on a structure, just add one of the "Auto" protocols:
+
+``` swift
+struct Contact, AutoJSONSerializable {
```
Comment (Liquidsoul) [inline README.md]: `,` should be `:` for protocol conformance.

**Review by Liquidsoul — CHANGES_REQUESTED** [review]
Thanks a lot @MonsieurDart ! 🚀 
Just some minor fixes here and there and it will be good to go!
If you don't have the time, I'll try to allocate some to do it in this weekend.

File: README.md (hunk 3 above)
Comment (MonsieurDart) [reply README.md]: Done in a following commit.

File: README.md (hunk 1 above)
Comment (MonsieurDart) [reply README.md]: Yup.

File: README.md (hunk 4 above)
Comment (MonsieurDart) [reply README.md]: Would be cool, indeed! 😊

File: README.md (hunk 6 above)
Comment (MonsieurDart) [reply README.md]: My bad… Fixed!

File: README.md (hunk 5 above)
Comment (MonsieurDart) [reply README.md]: Bien vu !

File: LICENSE (hunk 2 above)
Comment (MonsieurDart) [reply LICENSE]: As you wish!

**Comment by MonsieurDart** [comment]
Let me know if you have other stuff to fix. 😊

File: README.md [hunk 7]
```diff
@@ -114,12 +116,69 @@ struct Contact {
 }
 ```
 
+## Custom JSON*able Implementation ##
+
+If you want to nest some types that are not currently supported (for example: enums with associated values) or if you want to provide a special implementation of the serde methods for those types, you can simply implement the `JSONSerializable` and/or `JSONDeserializable` protocols.
```
Comment (Liquidsoul) [inline README.md:121]: Sorry I missed a typo, I suppose you meant `serialize` and not `serde` 😊

**Review by Liquidsoul — CHANGES_REQUESTED** [review]
Just one remaining typo in the docs I missed before and I'll merge this! 🚀

**Comment by MonsieurDart** [comment]
Serde is an (more or less) often used concatenation for "Serialize Deserialize." 😉

**Comment by Liquidsoul** [comment]
Didn't know that 😃
Let's merge that then!

**Review by Liquidsoul — APPROVED** [review]

**Pull request merged**